services/brig/src/Brig/App.hs

```diff
@@ -240,16 +239,17 @@ newEnv o = do
   let sett = Opt.optSettings o
   nxm <- initCredentials (Opt.setNexmo sett)
   twl <- initCredentials (Opt.setTwilio sett)
-  stomp <- case (Opt.stomp o, Opt.setStomp sett) of
-    (Nothing, Nothing) -> pure Nothing
-    (Just s, Just c) -> Just . Stomp.mkEnv s <$> initCredentials c
-    (Just _, Nothing) -> error "STOMP is configured but 'setStomp' is not set"
-    (Nothing, Just _) -> error "'setStomp' is present but STOMP is not configured"
-  -- This is messy. See Note [queue refactoring] to learn how we
-  -- eventually plan to solve this mess.
-  eventsQueue <- case Opt.internalEventsQueue (Opt.internalEvents o) of
-    StompQueue q -> pure (StompQueue q)
-    SqsQueue q -> SqsQueue <$> AWS.getQueueUrl (aws ^. AWS.amazonkaEnv) q
+  eventsQueue :: QueueEnv <- case Opt.internalEventsQueue (Opt.internalEvents o) of
+    StompQueueOpts q -> do
+      stomp :: Stomp.Env <- case (Opt.stomp o, Opt.setStomp sett) of
+        (Just s, Just c) -> Stomp.mkEnv s <$> initCredentials c
+        (Just _, Nothing) -> error "STOMP is configured but 'setStomp' is not set"
+        (Nothing, Just _) -> error "'setStomp' is present but STOMP is not configured"
+        (Nothing, Nothing) -> error "stomp is selected for internal events, but not configured in 'setStomp', STOMP"
+      pure (StompQueueEnv (Stomp.broker stomp) q)
+    SqsQueueOpts q -> do
+      let throttleMillis = fromMaybe Opt.defSqsThrottleMillis (view Opt.sqsThrottleMillis $ Opt.optSettings o)
+      SqsQueueEnv aws throttleMillis <$> AWS.getQueueUrl (aws ^. AWS.amazonkaEnv) q
   mSFTEnv <- mapM (Calling.mkSFTEnv sha512) $ Opt.sft o
```

Comment:
this is still messy but the reference the the `Note` is gone, I guess this is by accident?

Reply:
Hum, I think this is better now, and the referenced comment is also much shorter.

Reply:
alright, I guess I didn't understand the initial gripes with this function then